### PR TITLE
Bump wasm-bindgen

### PR DESCRIPTION
**Pull request by mwu-tow**
### Pull Request Description
This PR brings wasm-bindgen (and related crates) to the latest version.
I've also removed patching code, so future updates should be much easier.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

File: app/ide-desktop/lib/content/src/wasm_imports.js [hunk 1]
```diff
@@ -1,14 +1,3 @@
...
-/// https://github.com/rustwasm/wasm-pack/issues/790
-
-exports.wasm_imports = function () {
-    return wasm_rust_glue.default()
-}
-
-exports.after_load = function (w, module) {
-    return wasm_rust_glue.after_load(w, module)
-}
+exports.init = init
```
Comment (farmaazon) [inline app/ide-desktop/lib/content/src/wasm_imports.js:3]: Do we still need this file? Cannot we just import from wasm_rust_glue directly?

File: app/gui/view/graph-editor/src/component/visualization/foreign/java_script/binding.rs [hunk 2]
```diff
@@ -97,21 +97,33 @@ pub struct JsTheme {
...
+        /// The blue part as a float between 0 and 1
+        pub blue:  f32,
+        /// The opacity as a float between 0 and 1
+        pub alpha: f32,
+    }
 }
 
+// Make it as if module `js_color` members were defined here.
+pub use js_color::*;
+
```
Comment (farmaazon) [inline app/gui/view/graph-editor/src/component/visualization/foreign/java_script/binding.rs]: Is this comment needed? I think the code is self-explanatory.

File: Cargo.toml [hunk 3]
```diff
@@ -69,6 +69,9 @@ debug = true
...
 
 [workspace.dependencies]
-tokio = { version = "=1.20.1", features = ["full", "tracing"] }
 console-subscriber = "=0.1.7"
-nix = "=0.24.1" # DO NOT BUMP UNTIL NIGHTLY IS UPDATED. Otherwise, it brings too new libc.
+nix = "0.26.1"
+serde-wasm-bindgen = "0.4.5"
+wasm-bindgen = { version = "0.2.83", features = ["serde-serialize"] }
+wasm-bindgen-test = { version = "0.3.33" }
+tokio = { version = "=1.20.1", features = ["full", "tracing"] }
```
Comment (farmaazon) [inline Cargo.toml:77]: Do we have any policy what dependencies are kept here? Maybe we should put here every dependency (in another PR of course).

**Review by farmaazon — APPROVED** [review]

File: Cargo.toml (hunk 3 above)
Comment (mwu-tow) [reply Cargo.toml:77]: Not really, the feature was recently introduced. It makes much easier to manage dependencies versions (and prevent having the same dependency in two different versions), so I slowly move dependencies here when I touch them.

File: app/gui/view/graph-editor/src/component/visualization/foreign/java_script/binding.rs (hunk 2 above)
Comment (mwu-tow) [reply app/gui/view/graph-editor/src/component/visualization/foreign/java_script/binding.rs]: Ok, wwill remove.

File: app/ide-desktop/lib/content/src/wasm_imports.js (hunk 1 above)
Comment (mwu-tow) [reply app/ide-desktop/lib/content/src/wasm_imports.js:3]: No, this file serves as an entry point for the bundler.

**Pull request merged**